crypto/digestible/derive/test/tests/behavior.rs

```diff
@@ -67,38 +42,81 @@ struct GenericFoo<X: Digestible> {
 #[test]
 fn foo1() {
     let arg = Foo { a: 0, b: 1, c: 2 };
```

Comment:
It looks like a lot of the test data in this file is "benignly ordered", that is, the field names are often in alphabetical order and/or their values are in alphabetical order. It also seems like the field names are mostly/all single-letter names (i.e. don't contain underscores).

Seems like it would be good to diversify the test cases with more arbitrary field names and out-of-order fields and data.

Reply:
For nastier test cases, strings like "prim", "seq", or "agg" would make good field names or values.

Reply:
A field containing an empty Vec would be a good test case, too.

Reply:
done